tests/sentry/models/test_groupsnooze.py

```diff
@@ -79,54 +159,139 @@ def test_user_delta_reached(self):
 
     @mock.patch("django.utils.timezone.now")
     def test_user_rate_reached(self, mock_now):
-        mock_now.return_value = datetime(2016, 8, 1, 0, 0, 0, 0, tzinfo=timezone.utc)
+        """Test that ignoring an issue until it's hit by 100 users in an hour works."""
+        mock_now.return_value = self.now
 
-        snooze = GroupSnooze.objects.create(group=self.group, user_count=100, user_window=60)
-        tsdb.record(
-            tsdb.models.users_affected_by_group,
-            self.group.id,
-            [next(self.sequence) for _ in range(0, 101)],
+        event = self.store_event(
+            data={
+                "user": {"id": 999},
+                "timestamp": iso_format(before_now(minutes=2)),
+                "fingerprint": ["group1"],
+            },
+            project_id=self.project.id,
         )
+        snooze = GroupSnooze.objects.create(group=event.group, user_count=100, user_window=60)
+
+        for i in range(0, 100):
+            self.store_event(
+                data={
+                    "user": {"id": i},
+                    "timestamp": iso_format(self.now - timedelta(minutes=1)),
+                    "fingerprint": ["group1"],
+                },
+                project_id=self.project.id,
+            )
+
+        assert not snooze.is_valid(test_rates=True)
+
+    @mock.patch("django.utils.timezone.now")
+    def test_user_rate_reached_perf_issues(self, mock_now):
```

Comment:
I'm totally stuck on the perf tests here - I tried to leave comments to explain but basically I can only get them to pass if I use the `SnubaTSDB` rather than the Redis one (`from sentry import tsdb`). I haven't been able to figure out why, whether it's safe to replace that in production (we could use the Redis one for error issues and the Snuba one for perf issues, but that seems like a temporary solution), or how I can write the tests to work w/ the Redis one. 

Also, we have to change the start time for the user rate test (`start = end - timedelta(minutes=self.user_window + 1`) to pass, which is weird. Otherwise, not all of them will be found.

Also also, I'm capped at 6 transaction events for both tests. If I make any number higher than 6 they're never found from `tsdb.get_distinct_counts_totals` or `tsdb.get_sums`. I cannot figure out why.